compiler/noirc_evaluator/src/acir/arrays.rs

```diff
@@ -226,13 +226,6 @@ impl Context<'_> {
         }
 
         if let Some(store_value) = store_value {
-            // We should avoid storing a DynamicArray directly inside an Array. Instead we must allow
-            // the regular `array_set_value` to take place which reads the item from the block the
-            // dynamic array refers to, and writes items one by one. Otherwise subsequent operations
-            // with non-constant indexes will fail.
-            if matches!(store_value, AcirValue::DynamicArray(_)) {
-                return Ok(false);
-            }
```

Comment:
Just to confirm: the bailing out on the _read_ side cannot be solved in a similar way, right?